gridmenuview/src/androidTest/java/com/bcc/gridmenuview/MenuItemTest.java

```diff
@@ -5,44 +5,126 @@
 import androidx.test.platform.app.InstrumentationRegistry;
 
 import com.bcc.gridmenuview.model.MenuItem;
+import com.bcc.gridmenuview.provider.DrawableImageProvider;
+import com.bcc.gridmenuview.provider.LocalImageProvider;
+import com.bcc.gridmenuview.provider.NetworkImageProvider;
 
 import org.junit.Before;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
-public class MenuItemTest  {
+public class MenuItemTest {
+
     private MenuItem menuItem;
+    private MenuItem menuItemDrawable;
+    private MenuItem menuItemNetwork;
+    private MenuItem menuItemLocal;
+    private DrawableImageProvider drawableImageProvider;
+    private NetworkImageProvider networkImageProvider;
+    private LocalImageProvider localImageProvider;
 
     @Before
-    public void setUp()  {
-        menuItem = new MenuItem("Satu",InstrumentationRegistry
+    public void setUpMenuItemDrawable() {
+        menuItem = new MenuItem("Satu", InstrumentationRegistry
                 .getInstrumentation()
                 .getContext()
                 .getDrawable(R.drawable.ic_launcher_background));
+    }
+
+    @Before
+    public void setUpMenuItemDrawableImageProvider() {
+        drawableImageProvider = new DrawableImageProvider(InstrumentationRegistry
+                .getInstrumentation()
+                .getContext()
+                .getDrawable(R.drawable.ic_launcher_background));
+        menuItemDrawable = new MenuItem("Satu", drawableImageProvider);
+    }
+
+    @Before
+    public void setUpMenuItemNetworkImageProvider() {
+        networkImageProvider = new NetworkImageProvider(InstrumentationRegistry.getInstrumentation().getContext(), "https://cdn3.iconfinder.com/data/icons/capsocial-round/500/youtube3-128.png");
+        menuItemNetwork = new MenuItem("Satu", networkImageProvider);
+    }
 
+    @Before
```

Comment:
its totally fine to have multiple `@before` here but there is no guarantee which one run first 🤔

Reply:
is it better if i make another class for every menuItem test so each one have only one `@before`?